nbase/nbase_addrset.c

```diff
@@ -500,7 +500,7 @@ static int sockaddr_to_mask (const struct sockaddr *sa, int bits, u32 *mask)
       mask[i] = 0;
     }
     else {
-      mask[i] = ~((1 << (unmasked_bits - (32 * (4 - i)))) - 1);
+      mask[i] = ~(0xffffffff % (1 << unmasked_bits));
```

Comment:
This code suffers from a similar issue as the original one: There is no guaranteed outcome when `unmasked_bits` exceeds 31. As an example:
```C
#include <stdio.h>
void main ()
{
int m=32;
printf("%08x\n%08x\n",(0xFFFFFFFF<<(m-1)<<1),(0xFFFFFFFF<<m));
}
```
When compiled with GCC on Ubuntu 18.04 x64:
```
$ ./a.out 
00000000
ffffffff
```

Reply:
I assumed that it would hit another branch if `unmasked_bits` is a multiple of 32. But I my understanding of how `mask` is used in the rest of the code is limited, so I might miss something here.

Will test the other patch from the ticket on Monday